mlir/include/mlir/Target/LLVM/ROCDL/Utils.h

```diff
@@ -27,6 +27,62 @@ namespace ROCDL {
 /// 5. Returns an empty string.
 StringRef getROCMPath();
 
+/// Helper class for specifying the AMD GCN device libraries required for
+/// compilation.
+class AMDGCNLibraryList {
```

Comment:
Nit, isn't this just `enum class AMDGCNLibraryList : uint32_t`?

... Actually, on top of that - there's already general support for bit enums. This could and should be tablegen'd

Reply:
I was trying to avoid going to tablegen for such a small class, but I'll switch it.

Reply:
It's more that there's a lot of common infrastructure around these sorts of flag enums (including, say, printing them) that can be autogenerated. And it'll be useful to have if someone ever wants to put this as an attribute somewhere